packages/core/src/checkout-buttons/strategies/braintree/braintree-paypal-button-strategy.ts

```diff
@@ -155,19 +188,37 @@ export default class BraintreePaypalButtonStrategy implements CheckoutButtonStra
                 enableShippingAddress: true,
                 shippingAddressEditable: false,
                 shippingAddressOverride,
-                amount: checkout.outstandingBalance,
-                currency: config.currency.code,
+                amount,
+                currency: currencyCode,
                 offerCredit: false,
             });
         } catch (error) {
-            if (onError) {
-                onError(error);
+            if (onPaymentError) {
+                onPaymentError(error);
```

Comment:
Why did you change this one? As far as I see we are creating payment but not submitting it.
Please double check the description for the callback that you use:
https://github.com/bigcommerce/checkout-sdk-js/pull/1629/files#diff-718de5b34e6160523a486f1c1c5702d7449591d7ad1664d4e3ce1d7a79072289R38

Reply:
`onPaymentError` it is `onError` from `braintreepaypal`
I removed `onError` that we provide to `_setupPayment` and do destructurization there